### PR TITLE
[Feature][Executor] GPU Model Runner Supports prompt_logprobs and max_logprobs

### PR DESCRIPTION
**Pull request by ckl117**
## Motivation

GPUModelRunner supports max_logprobs=-1 and prompt_logprobs.
- computes vocal_size logprobs(including prompt_logprobs).
- computes prompt_logprobs(disable prefix cache).

## Usage or Command

```shell
export FD_USE_GET_SAVE_OUTPUT_V1=1

python -m fastdeploy.entrypoints.openai.api_server \
    --model ./ERNIE-4.5-0.3B-PT \
    --max-model-len 32768 \
    --max-num-seqs 128 \
    --tensor-parallel-size 1 \
    --enable-logprob \
    --max-logprobs -1 \
    --no-enable-prefix-caching \
```

## Accuracy Tests

TODO: Server layer should support `top_logprobs=-1` and `prompt_logprobs`. 

## Checklist

- [x] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [x] Format your code, run `pre-commit` before commit.
- [ ] Add unit tests. Please write the reason in this PR if no unit tests.
- [ ] Provide accuracy results.
- [ ] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: fastdeploy/worker/gpu_model_runner.py [hunk 1]
```diff
@@ -561,6 +565,8 @@ def insert_tasks_v1(self, req_dicts: List[Request], num_running_requests: int =
                     len(request.output_token_ids) if prefill_end_index >= len(input_ids) else 0
                 )
                 self.share_inputs["pre_ids"][idx : idx + 1] = -1
+                if request.sampling_params.prompt_logprobs is not None:
+                    self.prompt_logprobs_reqs[request.request_id] = request
```
Comment (yuanlehome) [inline fastdeploy/worker/gpu_model_runner.py]: 超长时间压测后有没有考虑内存增长的情况？

Comment (yuanlehome) [reply fastdeploy/worker/gpu_model_runner.py]: 另外RL场景中使用的话，需要在clear_requests函数中清空一下model_runner的一些对象，包括这个，也顺便梳理下有没有其他的对象需要清除

File: fastdeploy/worker/gpu_model_runner.py [hunk 2]
```diff
@@ -2694,3 +2705,70 @@ def prepare_rope3d(
...
         )
         return rope_emb_lst
+
+    def _get_prompt_logprobs_list(
+        self,
+        hidden_states: paddle.Tensor,
+    ) -> list[Optional[LogprobsTensors]]:
+        assert (
+            not self.fd_config.cache_config.enable_prefix_caching
+        ), "prompt_logprobs must disable prefix caching, --no-enable-prefix-caching."
```
Comment (gongshaotian) [inline fastdeploy/worker/gpu_model_runner.py]: check 逻辑能否移到 config.py 的 check func 里

Comment (ckl117) [reply fastdeploy/worker/gpu_model_runner.py]: prompt_logprobs预期是在客户端请求体里的，没法在服务启动时判断。

File: fastdeploy/worker/gpu_model_runner.py [hunk 3]
```diff
@@ -581,6 +587,8 @@ def insert_tasks_v1(self, req_dicts: List[Request], num_running_requests: int =
                 self.share_inputs["seq_lens_decoder"][idx : idx + 1] = 0
                 self.share_inputs["seq_lens_encoder"][idx : idx + 1] = 0
                 self.share_inputs["is_block_step"][idx : idx + 1] = False
+                self.prompt_logprobs_reqs.pop(request.request_id, None)
+                self.in_progress_prompt_logprobs.pop(request.request_id, None)
```
Comment (gongshaotian) [inline fastdeploy/worker/gpu_model_runner.py:592]: 抢占时不需要 del self.prompt_logprobs_reqs[req.request_id] 吗

Comment (ckl117) [reply fastdeploy/worker/gpu_model_runner.py:592]: 上面也有抢占时prompt_logprobs_reqs清除的逻辑啊。

File: fastdeploy/worker/gpu_model_runner.py [hunk 4]
```diff
@@ -2694,3 +2705,70 @@ def prepare_rope3d(
...
+                # (num_prompt_tokens - 1) tokens for this request in the prior
+                # step. There are no more prompt logprobs to produce.
+                continue
+            offset = self.share_inputs["cu_seqlens_q"][request.idx]
+            prompt_hidden_states = hidden_states[offset : offset + num_logits]
+            logits = self.model.compute_logits(prompt_hidden_states)
+            prompt_token_ids = request.prompt_token_ids[start_tok : start_tok + num_logits]
+            if isinstance(prompt_token_ids, np.ndarray):
+                prompt_token_ids = prompt_token_ids.tolist()
+            prompt_token_ids_tensor = paddle.to_tensor(prompt_token_ids, dtype="int64")
```
Comment (gongshaotian) [inline fastdeploy/worker/gpu_model_runner.py]: paddle 不支持 ndarray 直接转tensor吗

File: fastdeploy/worker/gpu_model_runner.py (hunk 1 above)
Comment (ckl117) [reply fastdeploy/worker/gpu_model_runner.py]: prompt变长和增大并发都会导致显存增长，但不存在显存泄露，这是符合预期的。

**Review by gongshaotian — APPROVED** [review]
LGTM

**Review by yuanlehome — APPROVED** [review]
